openproblems/tasks/spatial_decomposition/datasets/pancreas.py

```diff
@@ -3,53 +3,50 @@
 from ....tools.decorators import dataset
 from .utils import generate_synthetic_dataset
 
+import functools
 import scanpy as sc
 
-
-@dataset(
-    "Pancreas (alpha=1)",
+_pancreas_dataset = functools.partial(
+    dataset,
     data_url=load_pancreas.metadata["data_url"],
     data_reference=load_pancreas.metadata["data_reference"],
-    dataset_summary="Human pancreas cells aggregated from single-cell"
-    " (Dirichlet alpha=1)",
 )
-def pancreas_alpha_1(test=False, n_obs=100):
-    adata = load_pancreas(test=test, integer_only=True)
+_DATASET_SUMMARY = (
+    "Human pancreas cells aggregated from single-cell (Dirichlet alpha={})"
+)
+
+
+def _pancreas_synthetic(alpha: float, test: bool = False, n_obs: int = 100):
+    adata = load_pancreas(test=test, keep_techs=["inDrop3"])
```

Comment:
@LuckyMD this is the important change -- by adding this new API we can now specify which batch we want and still got 500 cells in test mode -- otherwise we filter after test mode reduction and will end up with <100 cells

Reply:
ah, this makes sense... thx

Reply:
would it not be better to have keep_techs as an argument to enable different batches to be used?

Reply:
Can do, yep.